From 0a24ecfe3ca2330888014c746312297f313314d6 Mon Sep 17 00:00:00 2001
From: Michael Bestas <mikeioannina@cyanogenmod.org>
Date: Tue, 9 Feb 2016 18:51:28 +0200
Subject: [PATCH 113/201] Snap: Fix crash when set infinite touch-focus
 duration

* java.lang.Integer.valueOf(string) method can't parse
  hexadecimal values

Change-Id: I42706db72fc9c1093a7aa16440af449f731e5176
---
 src/com/android/camera/PhotoModule.java | 2 +-
 src/com/android/camera/VideoModule.java | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/com/android/camera/PhotoModule.java b/src/com/android/camera/PhotoModule.java
index d643eeb84..0741f9761 100644
--- a/src/com/android/camera/PhotoModule.java
+++ b/src/com/android/camera/PhotoModule.java
@@ -4275,7 +4275,7 @@ public class PhotoModule
             }
 
             // Set focus time.
-            mFocusManager.setFocusTime(Integer.valueOf(
+            mFocusManager.setFocusTime(Integer.decode(
                     mPreferences.getString(CameraSettings.KEY_FOCUS_TIME,
                     mActivity.getString(R.string.pref_camera_focustime_default))));
         } else {
diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index ba9c6ac1e..e14fa1a97 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -2840,7 +2840,7 @@ public class VideoModule implements CameraModule,
         mParameters.setFocusMode(mFocusManager.getFocusMode(true));
 
         // Set focus time.
-        mFocusManager.setFocusTime(Integer.valueOf(
+        mFocusManager.setFocusTime(Integer.decode(
                 mPreferences.getString(CameraSettings.KEY_VIDEOCAMERA_FOCUS_TIME,
                 mActivity.getString(R.string.pref_camera_video_focustime_default))));
 
-- 
2.17.1

